rosbag2/src/rosbag2/writer.cpp

```diff
@@ -94,6 +120,23 @@ void Writer::write(std::shared_ptr<SerializedBagMessage> message)
     throw std::runtime_error("Bag is not open. Call open() before writing.");
   }
 
+  // Update the message count for the Topic.
+  ++topics_[message->topic_name].message_count;
```

Comment:
At should be used here, as I believe we should not implicitly create entries to this map, without calling create_topic first.

```suggestion
  ++topics_.at(message->topic_name).message_count;
```